From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Steinborn <git@steinborn.me>
Date: Fri, 13 Mar 2020 02:49:01 -0400
Subject: [PATCH] Add Bukkit#getConsoleSender()


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 8438831db02175f02991b67b1d7b5195c8e3b97f..040e51873701851502fa0e4d5d124e7a13c90517 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -260,4 +260,10 @@ public final class Bukkit {
     public static void reloadWhitelist() {
         server.reloadWhitelist();
     }
+
+    // Canyon start
+    public static org.bukkit.command.ConsoleCommandSender getConsoleSender() {
+        return server.getConsoleSender();
+    }
+    // Canyon end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 8b21ad20afcccfc07201b49673a0f2b18be1cf7b..dbea05bc38787629097ae747d1fddfd539730fd3 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -450,4 +450,12 @@ public interface Server {
      */
     public double[] getTPS();
     // Canyon end
+
+    // Canyon start
+    /**
+     * Returns a command sender representing the console
+     * @return a console command sender
+     */
+    org.bukkit.command.ConsoleCommandSender getConsoleSender();
+    // Canyon end
 }
