From b98ac783fa99f3fd0a5258da5c702a77be486537 Mon Sep 17 00:00:00 2001
From: Andrew Steinborn <git@steinborn.me>
Date: Fri, 13 Mar 2020 02:49:27 -0400
Subject: [PATCH] Implement Server#getConsoleSender()


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index e921c3bf..98e2521f 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -818,6 +818,13 @@ public final class CraftServer implements Server {
     }
     // Canyon end
 
+    // Canyon start
+    @Override
+    public ConsoleCommandSender getConsoleSender() {
+        return this.console.console;
+    }
+    // Canyon end
+
     public void setWhitelist(boolean value) {
         server.o = value;
         console.propertyManager.b("white-list", value);
-- 
2.43.0.windows.1

